### DISCUSSION TITLE
Output to HID USB ?

### DISCUSSION THREAD

**Discussion by schmurtzm** [darthcloud/BlueRetro#84] [Ideas]
Hi,

Is it possible to connect it to USB input of a PC / Raspberry / Mini Console ?
My idea is to use it on the Mega Drive Mini which has USB connectors.

Is it possible ?

(Thanks for this awesome project)
Best regards.

**Comment by darthcloud** [comment]
No USB, sorry. ESP32 has no USB PHY hardware.

**Comment by schmurtzm** [comment]
Esp32-s2 has hardware usb-otg ( host/device ) controller ,  it could be a lead  ? 😃  Some libraries already allows to emulate HID devices like keyboard like [this one](https://github.com/chegewara/EspTinyUSB)

There's also a software library for the classic ESP32  :  https://github.com/sdima1357/esp32_usb_soft_host

Or may be with the use of an additional arduino with the help of  V-USB  library to transfer the commands to a virtual USB gamepad ([like this for example](https://github.com/OpenRetroPad/OpenRetroPad) ?)

It could be awesome to have this possibility :p   
Once USB is managed it could offer compatibility with OG xbox , switch, mini consoles ... 🤩

**Comment by darthcloud** [comment]
The new ESP chip all drop support for Classic Bluetooth which means 95% of BT controller would not work anymore.

I'm aware of those soft USB implementation and Its likely to be a huge pain in this ass to end up with something half working.

However the #1 reason for this not happening is that it's a use case I personally do not need.

So sorry no USB

**Comment by schmurtzm** [comment]
Understood, thanks for the (fast) answer :)
